wezterm-gui/src/termwindow/render/screen_line.rs

```diff
@@ -349,8 +350,14 @@ impl crate::TermWindow {
                 + params.left_pixel_x
                 + (phys(params.cursor.x, num_cols, direction) as f32 * cell_width);
 
-            if cursor_shape.is_some() {
-                let mut quad = layers.allocate(0).context("layers.allocate(0)")?;
+            if let Some(shape) = cursor_shape {
+                let cursor_layer = match shape {
+                    CursorShape::BlinkingBar | CursorShape::SteadyBar => 2,
+                    _ => 0,
+                };
+                let mut quad = layers
+                    .allocate(cursor_layer)
+                    .context(format!("layers.allocate({cursor_layer})"))?;
```

Comment:
don't pay the cost of making the string unless this fails

```suggestion
                    .with_context(||format!("layers.allocate({cursor_layer})"))?;
```